src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -136,42 +145,53 @@ private SwitchBootstraps() {}
      *
      * @throws NullPointerException if any argument is {@code null}
      * @throws IllegalArgumentException if any element in the labels array is null, if the
-     * invocation type is not not a method type of first parameter of a reference type,
+     * invocation type is not a method type of first parameter of a reference type,
      * second parameter of type {@code int} and with {@code int} as its return type,
      * or if {@code labels} contains an element that is not of type {@code String},
-     * {@code Integer}, {@code Class} or {@code EnumDesc}.
+     * {@code Integer}, {@code Long}, {@code Float}, {@code Double}, {@code Boolean},
+     * {@code Class} or {@code EnumDesc}. Additionally, if {@code labels} contains an element
+     * that is not of type {@code Boolean} when {@code target} is a {@code Boolean.class}.
      * @jvms 4.4.6 The CONSTANT_NameAndType_info Structure
      * @jvms 4.4.10 The CONSTANT_Dynamic_info and CONSTANT_InvokeDynamic_info Structures
      */
     public static CallSite typeSwitch(MethodHandles.Lookup lookup,
                                       String invocationName,
                                       MethodType invocationType,
                                       Object... labels) {
+        Class<?> selectorType = invocationType.parameterType(0);
         if (invocationType.parameterCount() != 2
             || (!invocationType.returnType().equals(int.class))
-            || invocationType.parameterType(0).isPrimitive()
             || !invocationType.parameterType(1).equals(int.class))
             throw new IllegalArgumentException("Illegal invocation type " + invocationType);
         requireNonNull(labels);
 
         labels = labels.clone();
```

Comment:
just curious, why do we need to clone this array?

Reply:
No need. Remainings of a previous approach. Thanks for spotting